cpp/benchmarks/quantiles/tdigest.cpp

```diff
@@ -77,47 +72,113 @@ void bm_tdigest_merge(nvbench::state& state)
   tdigest_children.push_back(maxes.release());
   cudf::test::structs_column_wrapper tdigest(std::move(tdigest_children));
 
-  rmm::device_uvector<cudf::size_type> group_offsets(num_groups + 1, stream, mr);
-  rmm::device_uvector<cudf::size_type> group_labels(num_tdigests, stream, mr);
-  auto group_offset_iter = cudf::detail::make_counting_transform_iterator(
-    0,
-    cuda::proclaim_return_type<cudf::size_type>(
-      [tdigests_per_group] __device__(cudf::size_type i) { return i * tdigests_per_group; }));
-  thrust::copy(rmm::exec_policy_nosync(stream, mr),
-               group_offset_iter,
-               group_offset_iter + num_groups + 1,
-               group_offsets.begin());
-  auto group_label_iter = cudf::detail::make_counting_transform_iterator(
-    0,
-    cuda::proclaim_return_type<cudf::size_type>(
-      [tdigests_per_group] __device__(cudf::size_type i) { return i / tdigests_per_group; }));
-  thrust::copy(rmm::exec_policy_nosync(stream, mr),
-               group_label_iter,
-               group_label_iter + num_tdigests,
-               group_labels.begin());
+  // group offsets, labels
+  auto zero       = cudf::numeric_scalar<cudf::size_type>(0);
+  auto indices    = cudf::sequence(num_tdigests, zero);
+  auto tpg_scalar = cudf::numeric_scalar<cudf::size_type>(tdigests_per_group);
+
+  auto group_offsets = cudf::sequence(num_groups + 1, zero, tpg_scalar, stream, mr);
+  // expand 0, 1, 2, 3, 4, into 0, 0, 0, 1, 1, 1, 2, 2, 2, etc
+  auto group_labels = std::move(
+    cudf::repeat(cudf::table_view({cudf::slice(indices->view(), {0, num_groups}).front()}),
+                 tdigests_per_group,
+                 stream,
+                 mr)
+      ->release()
+      .front());
+
+  stream.synchronize();
 
   state.add_element_count(total_centroids);
 
   state.set_cuda_stream(nvbench::make_cuda_stream_view(stream.value()));
-  state.exec(nvbench::exec_tag::timer | nvbench::exec_tag::sync,
-             [&](nvbench::launch& launch, auto& timer) {
-               timer.start();
-               auto result = cudf::tdigest::detail::group_merge_tdigest(
-                 tdigest, group_offsets, group_labels, num_groups, max_centroids, stream, mr);
-               timer.stop();
-             });
+  state.exec(
+    nvbench::exec_tag::timer | nvbench::exec_tag::sync, [&](nvbench::launch& launch, auto& timer) {
+      timer.start();
+      auto result = cudf::tdigest::detail::group_merge_tdigest(
+        tdigest,
+        {group_offsets->view().begin<cudf::size_type>(),
+         static_cast<size_t>(group_offsets->size())},
+        {group_labels->view().begin<cudf::size_type>(), static_cast<size_t>(group_labels->size())},
```

Comment:
I believe `column_view` has a `device_span` operator so this may work
```suggestion
        group_offsets->view(),
        group_labels->view(),
```

Reply:
It's interesting.  I originally tried this, but it didn't work when it was a a cu file:

```
 error: no suitable user-defined conversion from "cudf::column_view" to "cudf::device_span<const cudf::size_type, 18446744073709551615UL>" exists
          group_offsets->view(), 
```

It does work as a cpp though.  nvcc definitely getting confused.